reactivesocket-core/src/main/java/io/reactivesocket/exceptions/Exceptions.java

```diff
@@ -28,31 +28,29 @@ private Exceptions() {}
 
     public static RuntimeException from(Frame frame) {
         final int errorCode = Frame.Error.errorCode(frame);
-        ByteBuffer dataBuffer = frame.getData();
-        String message = dataBuffer.remaining() == 0 ? "<empty message>" : ByteBufferUtil.toUtf8String(dataBuffer);
 
         RuntimeException ex;
         switch (errorCode) {
             case APPLICATION_ERROR:
-                ex = new ApplicationException(frame);
+                ex = new ApplicationException(new PayloadImpl(frame));
```

Comment:
You're right, it would probably be better to have a payload() method on the frame.